src/Features/Core/Portable/InvertConditional/AbstractInvertConditionalCodeRefactoringProvider.cs

```diff
@@ -3,23 +3,26 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Immutable;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CodeActions;
 using Microsoft.CodeAnalysis.CodeRefactorings;
 using Microsoft.CodeAnalysis.Editing;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
-using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.InvertConditional
 {
     internal abstract class AbstractInvertConditionalCodeRefactoringProvider<TConditionalExpressionSyntax>
-        : CodeRefactoringProvider
+        : SyntaxEditorBasedCodeRefactoringProvider
```

Comment:
This is the core change that would be needed in a CodeRefactoringProvider type to support FixAll. It needs to sub-type `SyntaxEditorBasedCodeRefactoringProvider` and implemented the following two abstract members:

1. `ImmutableArray<FixAllScope> SupportedFixAllScopes { get; }`
2. `Task FixAllAsync(Document document, ImmutableArray<TextSpan> fixAllSpans, SyntaxEditor editor, CancellationToken cancellationToken);`

Reply:
so is this change jsut for example pirposes?  if so, can we remove from this PR?

Reply:
(or did we have a strong belief that 'invert if' should be fix-allable?)

Reply:
This was one of the easier refactorings that to modify to add fix all support. One of the reasons was adding support to one of the refactorings in this PR is the ability to add both unit tests and integration tests. I agree this is not the most useful refactoring in terms of FixAll, but I don't see any harm in having it.